go/enclave/enclave.go

```diff
@@ -251,14 +252,26 @@ func (e *enclaveImpl) ExecuteOffChainTransaction(encryptedParams common.Encrypte
 	return e.chain.ExecuteOffChainTransaction(encryptedParams)
 }
 
-func (e *enclaveImpl) Nonce(address gethcommon.Address) uint64 {
-	// todo user encryption
+func (e *enclaveImpl) GetTransactionCount(encryptedParams common.EncryptedParamsGetTxCount) (common.EncryptedResponseGetTxCount, error) {
+	var nonce uint64
+	address, err := e.rpcEncryptionManager.ExtractAddress(encryptedParams)
+	if err != nil {
+		return nil, err
+	}
+
 	hs := e.storage.FetchHeadState()
-	if hs == nil {
-		return 0
+	if hs != nil {
+		// todo: we should return an error when head state is not available, but for current test situations with race
+		// 		conditions we allow it to return zero while head state is uninitialized
+		s := e.storage.CreateStateDB(hs.HeadRollup)
+		nonce = s.GetNonce(address)
+	}
+
+	encCount, err := e.rpcEncryptionManager.EncryptWithViewingKey(address, []byte(hexutil.EncodeUint64(nonce)))
```

Comment:
@joeldudley turned out this was the hexutil magic function I needed, found it after you mentioned prefixing on the "0x" to the string